--- a/ext/openssl/ossl_pkey_rsa.c
+++ b/ext/openssl/ossl_pkey_rsa.c
@@ -874,7 +874,11 @@
     rb_define_const(cRSA, "PKCS1_OAEP_PADDING", INT2NUM(RSA_PKCS1_OAEP_PADDING));
     rb_define_const(cRSA, "SSLV23_PADDING", INT2NUM(RSA_SSLV23_PADDING));
     rb_define_const(cRSA, "NO_PADDING", INT2NUM(RSA_NO_PADDING));
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
     rb_define_const(cRSA, "SSLV23_PADDING", INT2NUM(RSA_SSLV23_PADDING));
+#else
+    rb_define_const(cRSA, "SSLV23_PADDING", INT2NUM(RSA_PKCS1_PADDING));
+#endif
     rb_define_const(cRSA, "NO_PADDING", INT2NUM(RSA_NO_PADDING));
 #endif
     /* PKCS#1 v1.5 Encryption */